generated/nidaqmx/task.py

```diff
@@ -760,6 +760,46 @@ def read(self, number_of_samples_per_channel=NUM_SAMPLES_UNSET,
 
         return data.tolist()
 
+    def add_arguments_if_necessary(self, callback_function, expected_number_of_arguments):
+        if (callback_function.__code__.co_argcount < expected_number_of_arguments):
+            print("Not enough arguments! Adding arguments")
+            if ("every_n_samples_event_type" in parameters):
```

Comment:
I was actually trying the 2nd option specified in https://github.com/ni/nidaqmx-python/issues/143#:~:text=Make%20the%20callback,from%20internal_callback%20wrapper.

Basically my approach was to check if the users have specified task_handle or callback_data and if they didn't, then pass in those parameters as lambda parameters to the C function as done in the wrappers. That way, users can get away with only specifying the status / signal type / every n parameters in the callback function.

However I agree that this approach is heavily dependent on users correctly specifying the parameter names so I think your way of passing the task object directly and actually using the callback data would be better. I wasn't aware that python could do the `Optional[DONE_EVENT_CALLBACK]` so that approach is great. I will modify this PR to use your approach instead.